ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartPostHandler.java

```diff
@@ -0,0 +1,151 @@
+        Callback<ReadableStreamChannel> finalCallback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.finalCallback = finalCallback;
+      this.uri = restRequest.getUri();
+    }
+
+    /**
+     * Start the chain by calling {@link SecurityService#processRequest}.
+     */
+    private void start() {
+      try {
+        // Start the callback chain by performing request security processing.
+        securityService.processRequest(restRequest, securityProcessRequestCallback());
+      } catch (Exception e) {
+        finalCallback.onCompletion(null, e);
+      }
+    }
+
+    /**
+     * After {@link SecurityService#processRequest} finishes, call {@link SecurityService#postProcessRequest} to perform
+     * request time security checks that rely on the request being fully parsed and any additional arguments set.
+     * @return a {@link Callback} to be used with {@link SecurityService#processRequest}.
+     */
+    private Callback<Void> securityProcessRequestCallback() {
+      return buildCallback(frontendMetrics.putSecurityProcessRequestMetrics, securityCheckResult -> {
+        securityService.postProcessRequest(restRequest, securityPostProcessRequestCallback());
+      }, uri, LOGGER, finalCallback);
+    }
+
+    /**
+     * After {@link SecurityService#postProcessRequest} finishes, return response for the request.
+     * @return a {@link Callback} to be used with {@link SecurityService#postProcessRequest}.
+     */
+    private Callback<Void> securityPostProcessRequestCallback() {
+      return buildCallback(frontendMetrics.putSecurityPostProcessRequestMetrics, securityCheckResult -> {
+
+        if (restRequest.getArgs().containsKey(UPLOADS_PARAM_NAME)) {
+          // This is a request to initiate multipart upload request. Create an upload ID and return to user. This upload
+          // ID is specified in upload part requests and is used to associate all the parts in the specific multipart upload.
+          // Reference S3 API https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
+          String uploadId = UUID.randomUUID().toString();
+          RequestPath requestPath = (RequestPath) restRequest.getArgs().get(REQUEST_PATH);
+          NamedBlobPath namedBlobPath = NamedBlobPath.parse(requestPath, restRequest.getArgs());
+          String bucket = namedBlobPath.getAccountName();
+          String key = namedBlobPath.getBlobName();
+          LOGGER.debug("Sending response for CreateMultipartUpload. Bucket = {}, Key = {}, Upload Id = {}", bucket, key,
+              uploadId);
+          InitiateMultipartUploadResult initiateMultipartUploadResult = new InitiateMultipartUploadResult();
+          initiateMultipartUploadResult.setBucket(bucket);
+          initiateMultipartUploadResult.setKey(key);
+          initiateMultipartUploadResult.setUploadId(uploadId);
+          ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
Do we want to wrap all these to InitiateMultipartUploadResult.serialize() function?
```
          ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
          ObjectMapper objectMapper = new XmlMapper();
          objectMapper.writeValue(outputStream, initiateMultipartUploadResult);
```

Reply:
Based on Wei's suggestion, moved the Java POJO classes inside Handlers. May be we don't need separate serialize method now?